multisense_ros/src/ros_driver.cpp

```diff
@@ -121,7 +123,37 @@ int main(int argc, char** argvPP)
                                                        std::placeholders::_1),
                                              std::bind(&multisense_ros::Camera::groundSurfaceSplineDrawParametersChanged, &camera,
                                                        std::placeholders::_1));
-            ros::spin();
+
+            ros::Rate rate(50);
+            ros::Time last_status{ros::Time::now()};
+            const ros::Duration timeout{timeout_s};
+
+            ros::Time last_warning{};
+            ros::Duration warn_delay{0.5};
+            while (ros::ok()) {
+                ros::spinOnce();
+
+                system::StatusMessage statusMessage;
+                const auto status_result = d->getDeviceStatus(statusMessage);
+                if (Status_Ok != status_result) {
+
+                    if (ros::Time::now() - last_warning > warn_delay) {
+                        ROS_WARN("multisense_ros: lost connection with camera");
+                        last_warning = ros::Time::now();
+                    }
+
+                    if (timeout > ros::Duration{0} && ros::Time::now() - last_status > timeout) {
+                        ROS_ERROR("multisense_ros: shutting down due to connection timeout");
```

Comment:
Consider changing this to something like "reinitializing connection with the MultiSense"